packages/application-shell/src/configure-apollo.js

```diff
@@ -82,7 +84,10 @@ export const createApolloClient = () =>
         // However, a reference has the shape { typeId, id } where the id is the
         // id of the referenced entity. If we didn't prefix ids of References
         // they would clash with the referenced resource.
-        if (result.__typename === 'Reference')
+        // The same use case happens for DeliveryItem type in which the `id`
+        // is really the lineItem id referenced in the delivery
+        // https://docs.commercetools.com/http-api-projects-orders.html#deliveryitem
+        if (typeNameWithReferencedEntity.includes(result.__typename))
           return `${result.__typename}:${result.id}`;
```

Comment:
With this we assure that for the DeliveryItem type we got that specially mapped in the apollo cache 👍